src/components/tile-select/__internal__/accordion/accordion.component.tsx

```diff
@@ -1,19 +1,36 @@
 import React, { useEffect, useRef, useState } from "react";
-import PropTypes from "prop-types";
+import { SpaceProps } from "styled-system";
 
 import { StyledContentContainer, StyledContent } from "./accordion.style";
 import useResizeObserver from "../../../../hooks/__internal__/useResizeObserver";
 
-const Accordion = ({ children, expanded, contentId, controlId }) => {
+export interface AccordionProps extends SpaceProps {
+  children: React.ReactNode;
+  expanded?: boolean;
+  contentId?: string;
+  controlId?: string;
+}
+const Accordion = ({
+  children,
+  expanded,
+  contentId,
+  controlId,
+}: AccordionProps) => {
   const [contentHeight, setContentHeight] = useState(0);
-  const contentRef = useRef(null);
+  const contentRef = useRef<HTMLDivElement | null>(null);
 
   useResizeObserver(contentRef, () => {
-    setContentHeight(contentRef.current.scrollHeight);
+    // istanbul ignore else
+    if (contentRef.current) {
+      setContentHeight(contentRef.current.scrollHeight);
+    }
   });
 
   useEffect(() => {
-    setContentHeight(contentRef.current.scrollHeight);
+    // istanbul ignore else
+    if (contentRef.current) {
+      setContentHeight(contentRef.current?.scrollHeight);
```

Comment:
I think that we are already checking that in the conditional above
```suggestion
      setContentHeight(contentRef.current.scrollHeight);
```